lib/ecto/postgres/pg_utils.ex

```diff
@@ -0,0 +1,115 @@
+defmodule Vtc.Ecto.Postgres.Utils do
+  @moduledoc false
+
+  ## Exposes a macro for defining modules that will only be compiled if the caller
+  ## has set `:vtc, :postgres_types?` to `true` in their application config.
+
+  @spec __using__(Keyword.t()) :: Macro.t()
+  defmacro __using__(_) do
+    quote do
+      import Vtc.Ecto.Postgres.Utils
+
+      require Vtc.Ecto.Postgres.Utils
+    end
+  end
+
+  # Wraps defmodule, conditionally declaring the module during compilation
+  # based on caller configuration.
+  @spec defpgmodule(module(), do: Macro.t()) :: Macro.t()
+  defmacro defpgmodule(name, do: body) do
+    quote do
+      if Application.get_env(:vtc, :postgres_types?, false) do
+        :ok = unquote(__MODULE__).enforce_dep(Ecto, :ecto)
+        :ok = unquote(__MODULE__).enforce_dep(Postgrex, :postgrex)
+
+        defmodule unquote(name) do
+          unquote(body)
+        end
+      end
+    end
+  end
+
+  # Affirms that module from dep is present, throwing otherwise.
+  @spec enforce_dep(module(), atom()) :: :ok
+  def enforce_dep(module, name) do
+    if not Code.ensure_loaded?(module) do
+      throw(
+        "vtc: `:postgres_types?` config is true, but `#{module}` module not found. Add `#{name}` to your dependencies"
+      )
+    end
+
+    :ok
+  end
+
+  @typedoc """
+  Options type for `plpgsql_add_function/2`
+  """
+  @type create_func_opt() :: {:returns, atom()} | {:declare, Keyword.t(atom())} | {:body, String.t()}
+
+  @doc """
+  Builds a [plpgsql](https://www.postgresql.org/docs/current/plpgsql.html) function,
+  taking care of all the biolerplate.
+
+  ## Args
+
+  - `name`: The name of the function, including schema namespace.
+
+  ## Options
+
+  - `args`: The arguments the function takes and their types in a `arg: type` keyword
+    list.
+
+  - `returns`: The type the function returns.
+
+  - `declares`: A `name: type` keyword list of variables that should be declared in the
+    function's "DECLARES" block. Optionally can pass `name: {type, calculation}` to
+    declare a short calculation to set the variable.
+
+  - `body`: The function body.
+  """
+  @spec create_plpgsql_function(String.t(), [create_func_opt()]) :: String.t()
+  def create_plpgsql_function(name, opts) do
```

Comment:
It's plumbing for future Postgres types, like timecode and Framerate, so I just chucked it in a utility module since it will need to be shared pretty soon anyway.

Reply:
This module is set to @moduledoc false, so it won't show up in user-facing docs.

Reply:
generally, i'd prefer to see things like this as private first ...and then made public when they're actually needed (regardless of documentation/false, etc). it would be easy enough to make this public when the need arises (rather than future-proofing in this PR).

though, this may be moot if these functions are deferred to a later pr